src/main/java/com/amazon/aocagent/installers/otinstallers/OTPackageInstaller.java

```diff
@@ -38,57 +49,86 @@ private void downloadPackage() throws Exception {
     // get downloading link
     String s3Key = context.getTestingAMI().getS3Package().getS3Key(context.getAgentVersion());
     String downloadingLink =
-        "https://" + context.getStack().getS3BucketName() + ".s3.amazonaws.com/" + s3Key;
+            "https://" + context.getStack().getS3BucketName() + ".s3.amazonaws.com/" + s3Key;
 
     // get downloading command
     String downloadingCommand =
-        context
-            .getTestingAMI()
-            .getDownloadingCommand(
-                downloadingLink, context.getTestingAMI().getS3Package().getPackageName());
+            context
+                    .getTestingAMI()
+                    .getDownloadingCommand(
+                            downloadingLink,
+                            context.getTestingAMI().getS3Package().getPackageName());
 
     // execute downloading command
-    RetryHelper.retry(
-        () -> {
-          sshHelper.executeCommands(Arrays.asList(downloadingCommand));
-        });
+    if (context.getTestingAMI().getOSType().equals(OSType.WINDOWS)) {
+      ssmHelper.runShellScriptCommand(context.getInstanceId(), Arrays.asList(downloadingCommand),
+              GenericConstants.RUN_POWER_SHELL_SCRIPT_DOCUMENT.getVal());
+    } else {
+      RetryHelper.retry(
+          () -> {
+            sshHelper.executeCommands(Arrays.asList(downloadingCommand));
+          });
+    }
   }
 
   private void installPackage() throws Exception {
     // get installing command
     String installingCommand =
-        context
-            .getTestingAMI()
-            .getInstallingCommand(context.getTestingAMI().getS3Package().getPackageName());
+            context
+                    .getTestingAMI()
+                    .getInstallingCommand(context.getTestingAMI().getS3Package().getPackageName());
 
     // execute installing command
-    RetryHelper.retry(
-        () -> {
-          sshHelper.executeCommands(Arrays.asList(installingCommand));
-        });
+    if (context.getTestingAMI().getOSType().equals(OSType.WINDOWS)) {
```

Comment:
looks like there're multiple places we need to have this "if", so i feel it's better to have a new class for install ot package with ssm

Reply:
Not only the installation mechanism is different. The OS also matters. Windows is using ssm runPowershellCommand to install Agent. I believe Linux will use different document even if testing against ssm

Reply:
i think we can have one more method in the testami class like getssminstallcommand, so we don't need write "if" here. for linux we can leave it returning null for now